tests/tt_metal/tt_metal/test_pgm_dispatch_perf.cpp

```diff
@@ -0,0 +1,166 @@
+CoreRange workers_g = {{0, 0}, {0, 0}};;
+uint32_t size_g;
+uint32_t n_cbs_g;
+uint32_t n_args_g;
+bool brisc_enabled_g;
+bool ncrisc_enabled_g;
+bool trisc_enabled_g;
+
+void init(int argc, char **argv) {
+    std::vector<std::string> input_args(argv, argv + argc);
+
+    if (test_args::has_command_option(input_args, "-h") ||
+        test_args::has_command_option(input_args, "--help")) {
+        log_info(LogTest, "Usage:");
+        log_info(LogTest, "  -i: iterations", DEFAULT_ITERATIONS);
+        log_info(LogTest, "  -s: size of kernels in bytes (0..{})", MAX_KERNEL_SIZE);
+        log_info(LogTest, "  -x: X end of core range (1..x)");
+        log_info(LogTest, "  -y: Y end of core range (1..y)");
+        log_info(LogTest, "  -c: number of CBs (0..{})", MAX_CBS);
+        log_info(LogTest, "  -a: number of runtime args (0..{})", MAX_ARGS);
+        log_info(LogTest, "  -b: disable brisc kernel");
+        log_info(LogTest, "  -n: disable ncrisc kernel");
+        log_info(LogTest, "  -t: disable trisc kernels");
+        exit(0);
+    }
+
+    uint32_t core_x = test_args::get_command_option_uint32(input_args, "-x", 1);
+    uint32_t core_y = test_args::get_command_option_uint32(input_args, "-y", 1);
+    iterations_g = test_args::get_command_option_uint32(input_args, "-i", DEFAULT_ITERATIONS);
+    size_g = test_args::get_command_option_uint32(input_args, "-s", 1024);
+    n_cbs_g = test_args::get_command_option_uint32(input_args, "-c", 0);
+    n_args_g = test_args::get_command_option_uint32(input_args, "-a", 0);
+    if (size_g > MAX_KERNEL_SIZE) {
+        log_fatal("CB count must be 0..{}", MAX_KERNEL_SIZE);
+        exit(0);
+    }
+    if (n_cbs_g > MAX_CBS) {
+        log_fatal("CB count must be 0..{}", MAX_CBS);
+        exit(0);
+    }
+    if (n_args_g > MAX_ARGS) {
+        log_fatal("CB count must be 0..{}", MAX_ARGS);
+        exit(0);
+    }
+
+    brisc_enabled_g = !test_args::has_command_option(input_args, "-b");
+    ncrisc_enabled_g = !test_args::has_command_option(input_args, "-n");
+    trisc_enabled_g = !test_args::has_command_option(input_args, "-t");
+
+    workers_g = {.start = {0, 0}, .end = {core_x, core_y}};
+}
+
+int main(int argc, char **argv) {
+    init(argc, argv);
+
+    tt::llrt::OptionsG.set_kernels_nullified(true);
+
+    bool pass = true;
+    try {
+        int device_id = 0;
+        tt_metal::Device *device = tt_metal::CreateDevice(device_id);
+
+        CommandQueue& cq = tt::tt_metal::detail::GetCommandQueue(device);
+
+        tt_metal::Program program = tt_metal::CreateProgram();
+
+        std::map<string, string> pad_defines = {
+            {"KERNEL_BYTES", std::to_string(size_g)}
+        };
+
+        vector<uint32_t> args;
+        args.resize(n_args_g);
+
+        for (int i = 0; i < n_cbs_g; i++) {
+            tt_metal::CircularBufferConfig cb_config = tt_metal::CircularBufferConfig(16, {{i, tt::DataFormat::Float16_b}})
+                .set_page_size(i, 16);
```

Comment:
This page size corresponds to the size of the page data, not the size of the config. I guess in our case, it can be anything since the kernels don't actually run.